src/lightning_app/core/app.py

```diff
@@ -307,7 +308,10 @@ def maybe_apply_changes(self) -> bool:
         if not deltas:
             # When no deltas are received from the Rest API or work queues,
             # we need to check if the flow modified the state and populate changes.
-            if Delta(DeepDiff(self.last_state, self.state, verbose_level=2)).to_dict():
+            delta = Delta(DeepDiff(self.last_state, self.state, verbose_level=2))
+            difference = delta.to_dict()
```

Comment:
I would rework as:
```python
if not deltas:
    if DeepDiff(self.last_state, self.state, verbose_level=2):
        self.set_state(self.state)
        self._has_updated = True

    return False
```
since `DeepDiff` is truthy when there are differences and falsey otherwise.

Reply:
Done !